Backend/app/dashboard/Lib/src/Domain/Types/Transaction.hs

```diff
@@ -46,6 +46,20 @@ data Transaction = Transaction
     createdAt :: UTCTime
   }
 
+data TransactionMultipleRides = TransactionMultipleRides
+  { id :: Id Transaction,
+    requestorId :: [Id DP.Person],
+    serverName :: Maybe DSN.ServerName,
+    merchantId :: Maybe (Id DM.Merchant), -- will be Nothing for admin apis
+    endpoint :: Endpoint,
+    commonDriverIds :: [Id Common.Driver],
+    commonRideIds :: [Id Common.Ride],
+    request :: Maybe Text,
+    response :: Maybe Text,
+    responseError :: Maybe Text,
+    createdAt :: UTCTime
+  }
```

Comment:
We can also use datakinds to handle minor type changes in case like this together. But, I think it should be fine for now to not make it complex. Also, this should go as one transaction so the task would lose its meaning, if we go and change other APIs. Mb in next iteration, we can unify these and have the transaction for multiple rides only(we can send one element if we have one rideId). 
cc : @roman-bodavskiy

Reply:
@0utkarsh @mohdshoaib18169 I don't think we should create separate type here or have some complex logic with datakinds. Instead we can:

1. Save ` RideSyncReq` as `request` in transaction table and put `commonDriverId = Nothing` and `commonRideId = Nothing` (the same as we have in `stuckBookingsCancel` endpoint)
2. Or create list of ordinary transactions:
```
buildTransactionMutipleRides ::
  ( MonadFlow m,
    Common.HideSecrets request
  ) =>
  DT.Endpoint ->
  Maybe ServerName ->
  Maybe ApiTokenInfo ->
  [Id Common.Driver] ->
  [Id Common.Ride] ->
  Maybe request ->
  m [DT.Transaction]
```

First solution I like more, because I think it would be correct if one api call will save only one transaction, not list

Reply:
fixed it